backend/owa.ts

```diff
@@ -33,26 +46,43 @@ export async function fetchSessionData(partition: string, url: string, interacti
       };
       let checkLoginFinished = async function() {
         try {
-          response = await session.fetch(url + 'sessiondata.ashx', { method: 'POST' });
+          response = await session.fetch(urlObj.toString() + 'sessiondata.ashx', { method: 'POST' });
           finish(await response.json());
         } catch (ex) {
         }
       }
-      let checkCanary = function(_event, cookie, _cause, removed) {
-        if (!removed && cookie.name == 'X-OWA-CANARY') {
-          checkLoginFinished();
+      let checkCanary = async function(_event, cookie, _cause, removed) {
```

Comment:
Let's rename the function while we're at it (you also need to adapt the caller):

```suggestion
      let onCookie = async function(_event, cookie, _cause, removed) {
```